src/Service.GraphQLBuilder/Mutations/CreateMutationBuilder.cs

```diff
@@ -87,50 +230,59 @@ public static class CreateMutationBuilder
             }
 
             HotChocolate.Language.IHasName? definition = definitions.FirstOrDefault(d => d.Name.Value == field.Type.NamedType().Name.Value);
+
             // When creating, you don't need to provide the data for nested models, but you will for other nested types
-            // For cosmos, allow updating nested objects
-            if (definition != null && definition is ObjectTypeDefinitionNode objectType && IsModelType(objectType) && databaseType is not DatabaseType.CosmosDB_NoSQL)
+            if (definition != null && definition is ObjectTypeDefinitionNode objectType && IsModelType(objectType))
             {
-                return false;
+                return DoesRelationalDBSupportNestedCreate(databaseType);
             }
 
-            return true;
+            return false;
+        }
+
+        /// <summary>
+        /// Helper method to check if a field in an entity(table) is a referencing field to a referenced field
+        /// in another entity.
+        /// </summary>
+        /// <param name="field">Field definition.</param>
+        private static bool DoesFieldHaveReferencingFieldDirective(FieldDefinitionNode field)
+        {
+            return field.Directives.Any(d => d.Name.Value.Equals(ReferencingFieldDirectiveType.DirectiveName));
         }
 
-        private static InputValueDefinitionNode GenerateSimpleInputType(NameNode name, FieldDefinitionNode f)
+        /// <summary>
+        /// Helper method to create input type for a scalar/column field in an entity.
+        /// </summary>
+        /// <param name="name">Name of the field.</param>
+        /// <param name="fieldDefinition">Field definition.</param>
+        /// <param name="databaseType">Database type</param>
+        private static InputValueDefinitionNode GenerateScalarInputType(NameNode name, FieldDefinitionNode fieldDefinition, DatabaseType databaseType)
         {
             IValueNode? defaultValue = null;
 
-            if (DefaultValueDirectiveType.TryGetDefaultValue(f, out ObjectValueNode? value))
+            if (DefaultValueDirectiveType.TryGetDefaultValue(fieldDefinition, out ObjectValueNode? value))
             {
                 defaultValue = value.Fields[0].Value;
             }
 
+            bool isFieldNullable = defaultValue is not null ||
+                (!IsNoSQLDb(databaseType) && DoesRelationalDBSupportNestedCreate(databaseType) && DoesFieldHaveReferencingFieldDirective(fieldDefinition));
+
             return new(
                 location: null,
-                f.Name,
-                new StringValueNode($"Input for field {f.Name} on type {GenerateInputTypeName(name.Value)}"),
-                defaultValue is not null ? f.Type.NullableType() : f.Type,
+                fieldDefinition.Name,
+                new StringValueNode($"Input for field {fieldDefinition.Name} on type {GenerateInputTypeName(name.Value)}"),
+                defaultValue is not null ||
```

Comment:
why is this condition `defaultValue is not null` still here if `isFieldNullable` already includes it?